### PR TITLE
chore: Replace some `map().unwrap_or()` with `map_or()`.

### PR DESCRIPTION
**Pull request by stiegerc**

### CODE REVIEW

**Comment by stiegerc** [comment]
- There is no `map_or_default()` for some reason. You could write it as `map_or_else(Default::default, |_| ..)` but I don't think it's better.
- There are also such options for `map_err().unwrap_or()` as in `map_err_or()`.

So FWIW I replaced some low hanging fruit in crates owned by the message routing team.

File: rs/xnet/payload_builder/src/lib.rs [hunk 1]
```diff
@@ -893,8 +892,7 @@ pub fn get_msg_limit(subnet_id: SubnetId, state: &ReplicatedState) -> Option<usi
                     .network_topology
                     .subnets
                     .get(&subnet_id)
-                    .map(|subnet| subnet.subnet_type)
-                    .unwrap_or(Application); // Technically unwrap() would work here, but this is safer.
+                    .map_or(Application, |subnet| subnet.subnet_type); // Technically unwrap() would work here, but this is safer.
```
Comment (schneiderstefan) [inline rs/xnet/payload_builder/src/lib.rs:895]:
```suggestion
                    .map_or(Application, |subnet| subnet.subnet_type); // Technically map().unwrap() would work here, but this is safer.
```

**Review by schneiderstefan — APPROVED** [review]
LGTM. Did you find them manually or does clippy warn about them?

**Review by alexandru-uta — APPROVED** [review]

**Comment by stiegerc** [comment]
> LGTM. Did you find them manually or does clippy warn about them?

No I just discovered `map_or()` some weeks ago. This is my personal vendetta against `map().unwrap_or()` since it's just needlessly ugly :)

**Review by alin-at-dfinity — APPROVED** [review]
Agreed that `.map(...).unwrap_or_else()` does not look great. But IMHO neither does `.a_or_b(b, a)`.

I mean, why?...

(Just ranting about the Rust peaople's decisions, not your PR.)

**Comment by stiegerc** [comment]
> Agreed that `.map(...).unwrap_or_else()` does not look great. But IMHO neither does `.a_or_b(b, a)`.
> 
> I mean, why?...
> 
> (Just ranting about the Rust peaople's decisions, not your PR.)

I was a bit surprised by that also. I would have expected `.a_or_b(a, b)` but I guess it goes with the flow of reading it.

**Pull request merged**